frontend/src/pages/modelRegistry/screens/ModelVersionDetails/ModelVersionRegisteredFromLink.tsx

```diff
@@ -0,0 +1,174 @@
+import React from 'react';
+import { Button, Content, Flex, FlexItem, Popover, Stack, StackItem } from '@patternfly/react-core';
+import { Link } from 'react-router-dom';
+import { OutlinedQuestionCircleIcon } from '@patternfly/react-icons';
+import TypedObjectIcon from '~/concepts/design/TypedObjectIcon';
+import { ProjectObjectType } from '~/concepts/design/utils';
+import { ProjectsContext } from '~/concepts/projects/ProjectsContext';
+import { FindAdministratorOptions } from '~/pages/projects/screens/projects/const';
+import PopoverListContent from '~/components/PopoverListContent';
+import { globalPipelineRunDetailsRoute, getCatalogModelDetailsRoute } from '~/routes';
+import { ModelArtifact } from '~/concepts/modelRegistry/types';
+import {
+  modelSourcePropertiesToCatalogParams,
+  modelSourcePropertiesToPipelineRunRef,
+} from '~/concepts/modelRegistry/utils';
+
+/**
+ * Props for the ModelVersionRegisteredFromLink component
+ */
+type ModelVersionRegisteredFromLinkProps = {
+  modelArtifact: ModelArtifact;
+};
+
+/**
+ * Component that displays where a model version was registered from
+ * Uses utility functions to determine if it was from a pipeline run or catalog model
+ */
+const ModelVersionRegisteredFromLink: React.FC<ModelVersionRegisteredFromLinkProps> = ({
+  modelArtifact,
+}) => {
+  const { projects } = React.useContext(ProjectsContext);
+
+  // Extract information using utility functions
+  const registeredFromCatalogDetails = modelSourcePropertiesToCatalogParams(
+    modelArtifact.modelSourceKind,
+    modelArtifact.modelSourceClass,
+    modelArtifact.modelSourceGroup,
+    modelArtifact.modelSourceName,
+    modelArtifact.modelSourceId,
+  );
+
+  const registeredFromPipelineDetails = modelSourcePropertiesToPipelineRunRef(
+    modelArtifact.modelSourceKind,
+    modelArtifact.modelSourceGroup,
+    modelArtifact.modelSourceId,
+    modelArtifact.modelSourceName,
+  );
```

Comment:
If you change these utilities to use a single argument with a `ModelSourceProperties` type, you can just pass in the `modelArtifact` here since it'll be assignable to that type.